File: src/main/java/dev/interfiber/karpet/server/utils/SpawnLocator.java
package dev.interfiber.karpet.server.utils;

import net.minestom.server.instance.InstanceContainer;
import net.minestom.server.tag.Tag;
import org.jglrxavpok.hephaistos.nbt.NBTCompound;

/**
 *
 * @author persephone
 */
public class SpawnLocator {
    
    public static SpawnLocation GetSpawnLocation(InstanceContainer instance) {
            NBTCompound Compound = (NBTCompound) instance.getTag(Tag.NBT("Data"));
            Integer SpawnX = Compound.getInt("SpawnX");
            Integer SpawnY = Compound.getInt("SpawnY");
            Integer SpawnZ = Compound.getInt("SpawnZ");
            SpawnLocation Location = new SpawnLocation();
            if (SpawnX != null && SpawnY != null && SpawnZ != null) {
                Location.SpawnX = SpawnX;
                Location.SpawnY = SpawnY;
                Location.SpawnZ = SpawnZ;
            } else {
                System.out.println("Util error: Failed to read NBT data!");
                Location.SpawnX = 0;
                Location.SpawnY = 200;
                Location.SpawnZ = 0;
            }
            return Location;
    }
}

File: src/main/kotlin/dev/interfiber/karpet/server/utils/SpawnLocator.kt
package dev.interfiber.karpet.server.utils

import net.minestom.server.instance.InstanceContainer
import net.minestom.server.tag.Tag
import org.jglrxavpok.hephaistos.nbt.NBTCompound


class SpawnLocator {
    fun getSpawnLocation(instance: InstanceContainer): SpawnLocation? {
        val compound = instance.getTag(Tag.NBT("Data")) as NBTCompound
        val spawnX = compound.getInt("SpawnX")
        val spawnY = compound.getInt("SpawnY")
        val spawnZ = compound.getInt("SpawnZ")
        val location = SpawnLocation()
        if (spawnX != null && spawnY != null && spawnZ != null) {
            location.spawnX = spawnX.toDouble()
            location.spawnY = spawnY.toDouble()
            location.spawnZ = spawnZ.toDouble()
        } else {
            println("Util error: Failed to read NBT data!")
            location.spawnX = 0.0
            location.spawnY = 200.0
            location.spawnZ = 0.0
        }
        return location
    }
}